### PR TITLE
Feature/step animation

### PR DESCRIPTION
**Pull request by mycheckgithub**
Step animation support for baked/bezier export option
Add <step> marker with technique marker inside <Animation>

As collada specification 1.5 only support STEP semantic and not STEP_NEXT (like in maya)
<step> has been added to tell us if STEP is a "STEP" or a "STEP_NEXT"

in the case of bezier export option:

<Name_array id="obj-interpolation-array" count="4">BEZIER STEP BEZIER STEP </Name_array>
 <technique profile="OpenCOLLADAMaya">
     <step>STEP_NEXT STEP </step>
 </technique>

the second interpolation type will be a "STEP_NEXT" 
the 4th interpolation type will be a "STEP"

in the case of baked export option:

All transformations are stacked so we need to know if one "STEP" interpolation at time T is used for 
translationX or/and translationY or/and translationZ or/and RotationX or/and RotationY or/and RotationZ or/and Scale X or/and Scale Y or/and Scale Z

<Name_array id="obj-interp-array" count="5">BEZIER STEP STEP BEZIER STEP </Name_array>
 <technique profile="OpenCOLLADAMaya">
     <step>STEPNEXT_TX 
                STEP_RZ|STEP_RY|STEP_RX|STEP_TZ|STEP_TY            
                STEP_RZ|STEP_RY|STEP_RX|STEP_TZ|STEP_TY|STEP_TX 
     </step> 
 </technique>

In this case:
the 2nd interpolation is "STEP", it is a step for TranslationX and it's a "STEP_NEXT"
the 3rd interpolation is  "STEP", it is a step for RotationX & Y & Z and Translation Y & Z
the 5th interpolation is "STEP", it is a step for RotationX & Y & Z and Translation X & Y & Z

step is written like this:
1/  "STEP" or "STEPNEXT" 
2/ "_" + "TX" or "TY" or "TZ" or "RX" or "RY" or "RZ" if needed (only for baked export option)
3/ "|" is used to concatenate many different steps track if needed (only for baked export option)

During Fl4re animation import,

-  if the animation is a baked one, we can recreate matrix with all information from <Step> marker and add it at the right frame.
2 matrix frames are generated at the same time T (timing of the step) one with the value just before the step and the other with the value just after the step. 
So we can play animation only with interpolation between frames and step frame will be supported (by adding 2 frames at the time of the step)

- if the animation is a bezier one, during baking/compression inside Fl4re, all informations from <Step> marker are used to add 2 frames at the time of the step for each track independently

### CODE REVIEW

**Comment by RemiArnaud** [comment]
add to whitelist

File: COLLADAMaya/src/COLLADAMayaAnimationExporter.cpp [hunk 1]
```diff
@@ -1004,6 +1022,82 @@ namespace COLLADAMaya
...
+				break;
+			case MEulerRotation::kZXY:
+				FinalOrder = "YXZ";
+				break;
+			case MEulerRotation::kZYX:
+				FinalOrder = "XYZ";
+				break;
+			default:
+				// Export XYZ euler rotation in Z Y X order in the file.
+				// The rotation order is set to XYZ, Collada reads the parameter from behind.
```
Comment (RemiArnaud) [inline COLLADAMaya/src/COLLADAMayaAnimationExporter.cpp]: do you mean backwards ?

**Comment by RemiArnaud** [comment]
@hobbes1069 @aothms   -  bringing this to your attention.
Please let us know if this breaks compatibility, so we can fix it there before this gets merged in master.

Thanks!

File: COLLADAMaya/src/COLLADAMayaAnimationExporter.cpp (hunk 1 above)
Comment (mycheckgithub) [inline COLLADAMaya/src/COLLADAMayaAnimationExporter.cpp]: This is exactly done in this order in ColladaMayaRotateHelper.cpp

```javascript
 void RotateHelper::createRotation ( )
    {
        switch (rotation.order)
        {
        case MEulerRotation::kXYZ: 
            createZ(0); createY(1); createX(2); 
            break;

    ...
}
```

This code will add a new technique into source channel.

```javascript
 <technique profile="OpenCOLLADAMaya">
          <step>STEP_RZ|STEP_RY|STEP_RX|STEP_TZ|STEP_TY|STEPNEXT_TX STEP_RZ|STEP_RY|STEP_RX|STEP_TZ|STEP_TY|STEP_TX </step>
          <order>ZYX </order>
        </technique>
```

Comment (mycheckgithub) [inline COLLADAMaya/src/COLLADAMayaAnimationExporter.cpp]: This PR has been added into fl4re fork 
issue https://github.com/fl4re/OpenCOLLADA/issues/5

Comment (RemiArnaud) [inline COLLADAMaya/src/COLLADAMayaAnimationExporter.cpp]: "from behind" (= par derrière) should be replaced by something more meaningful... feel free to copy/paste the above information as well as comment in the code.

thx!

**Pull request merged**